### ISSUE TITLE
Parsing Japanese series

### ISSUE THREAD

**Issue by meapwork** [Kareadita/Kavita#1363]
**Describe the bug**
Since the last update added support for parsing Japanese volume number for individual volumes I think it makes sense to have it parse the series names the same way as well.

2 common ways to name series are as follow (folder names):

Ya Boy Kong Ming Vol1-Vol9: パリピ孔明 第01-09巻
Slam Dunk All 31 volumes: スラムダンク 全31巻

**To Reproduce**
1. Have the parent folder renamed to パリピ孔明 第01-09巻 or スラムダンク 全31巻
2. Scan library in Kavita
4. See errors: - Kavita will parse パリピ孔明 第01-09巻 as パリピ孔明 第01 and group files inside the folder properly into a single series (as long as the files inside are named uniformly eg: パリピ孔明 第01巻.zip...)
5. Kavita will parse all files in side スラムダンク 全31巻 as individuals series (all 31 files eg: スラムダンク 第01巻.zip will be turn into 31 different series)

**Expected behavior**
パリピ孔明 第01-09巻
スラムダンク 全31巻

Strip 第01-09巻 and 全31巻 from the titles and have it parse 2 series as パリピ孔明 and スラムダンク

If we go further:
第01-09巻 indicates the series is still Ongoing and 9 is the latest volume
全31巻 indicates the series is Completed and 31 is the final volume

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: Chrome
 - Version: Version 103.0.5060.66 (Official Build) (64-bit)

**Comment by majora2007** [comment]
The last update didn't add support for parsing Japanese kanji based series titles. If you want me to try, you need to provide a big list of different examples, like:

Accel world - vol 1 ch 3 -> accel world, vol 1, chapter 3.

But obviously in Japanese characters. Without that, I can't offer any support.

**Comment by meapwork** [comment]
![dlr_k](https://user-images.githubusercontent.com/107278763/179368120-de42e724-ede9-4c3b-955d-f3e80ad80756.png)

Thanks for replying. Sorry if I didn't make it clear but that was specifically for requesting support for parsing Japanese kanji based series titles as you said. I've attached a screenshot above which has the naming convention of a popular download site. 
(e.g. Say I have a folder for One Piece named ワンピース 第01-102巻)

Basically the solution is to just strip out everything "第number-number巻" and "全number巻" from the series titles similar to Kavita's current handling of parentheses.

**Comment by majora2007** [comment]
Hey, I need actual text and filenames in order to attempt to support this. An image isn't good because I can't type kanji nor does it encapsulate what Kavita is likely to find on disk. Hope you can provide examples so I can enhance support.

**Comment by meapwork** [comment]
Ah I thought that was clear sorry. So my current structure is like this, 
Taking One Piece, an ongoing series as an example (ワンピース is for One Piece):

Folder: ワンピース 第01-102巻
Filename: 
ワンピース 第01巻.rar -> ワンピース Volume 1
...
ワンピース 第102巻.rar -> ワンピース Volume 102

And for Naruto, a completed series (ナルト is for Naruto)
Folder: ナルト 全72巻
Filename (same structure as One Piece): 
ナルト 第01巻.rar -> ナルト Volume 1
...
ナルト 第72巻.rar -> ワンピース Volume 72

I don't have them by chapters, only by volumes. Hope it makes sense.

**Comment by majora2007** [comment]
This definitely helps. Do you have non clean filenames? These are pristine filenames, but usually people have things with underscores, extra tags (like who scanned them, etc).

When I build out parsing support, I need these non-pristine to improve the adaptability in other situations.

**Comment by meapwork** [comment]
Yeah I always have to use renamer tools to make the filenames as clean as possible to avoid any parsing errors. But most of time it's pretty straightforward with the volume number being closed between the aforementioned 2 characters  (第number巻). 
In case you're wondering, 第 is the prefix for number and 巻 means volume.

For the list, I just looked up a bunch of top downloaded files and they're as followed:

鬼滅の刃 19 [aKraa].cbz **->** 鬼滅の刃 Vol. 19
(Note: I know that Kavita will parse 19 as chapter 19 but in this case I think that's on the users to rename the files accordingly for it to parse correctly, this example can be ignored)

[諌山創] 進撃の巨人 第23巻.rar **->** 進撃の巨人 Vol. 23
(Note: [諌山創] here in the square brackets is the author's name: 諌山創, I actually name my folders this way as well but since Kavita ignores everything in parentheses it doesn't matter much)

The exhaustive list:

鬼滅の刃 19 [aKraa].cbz **->** 鬼滅の刃 Vol. 19
[諌山創] 進撃の巨人 第23巻.rar **->** 進撃の巨人 Vol. 23
(一般コミック) [奥浩哉] いぬやしき 第09巻 **->** いぬやしき Vol. 09
Gotoubun_No_Hanayome_v08.rar **->** Gotoubun_No_Hanayome Vol. 08
無職転生 ~異世界行ったら本気だす~ 第09巻.rar **->** 無職転生 ~異世界行ったら本気だす~ Vol. 09
The Promised Neverland v02.rar (many files are named this way with "v" similar to Comics)
Kingdom_v54.rar **->** Kingdom Vol. v54
[白井カイウ×出水ぽすか] 約束のネバーランド 第04巻.rar **->** 約束のネバーランド Vol. 04
(一般コミック) [伏瀬×川上泰樹] 転生したらスライムだった件 第14巻 [Tensei Shitara Slime Datta Ken vol 14] **->** 転生したらスライムだった件 Vol. 14
Kimetsu_No_Yaiba_v15.rar **->** Kimetsu_No_Yaiba Vol.15
キングダム -KINGDOM- 第01巻.zip **->** キングダム -KINGDOM- Vol. 01
(一般コミック)[春場ねぎ]五等分の花嫁 第14巻.zip **->** 五等分の花嫁 Vol.14
[南勝久] ザ・ファブル 第01巻 [waifu処理／h1280].rar **->** ザ・ファブル Vol. 01
[南勝久] ザ・ファブル 第18巻jpg+fix.rar **->** ザ・ファブル Vol. 18
[尾田栄一郎] ONE PIECE 第94巻  **->** ONE PIECE Vol. 94
Gotoubun_No_Hanayome_v10.rar **->** Gotoubun_No_Hanayome v10
[冨樫義博] HUNTER×HUNTER ハンター×ハンター 第36巻.rar **->** HUNTER×HUNTER ハンター×ハンター Vol. 36
[石田スイ] 東京喰種トーキョーグールre 第12巻.rar **->** 東京喰種トーキョーグールre Vol. 12
=====

I don't actually have issues with how Kavita deals with filenames just how it parses series names like if I have a folder named パリピ孔明 第01-09巻 it will parse the series name as パリピ孔明 第01 which doesn't make any sense (details in my original post above). Hope you can figure out some new patterns based on these.

**Comment by majora2007** [comment]
This is exactly what I'm looking for. I'll take a look this release and hopefully can improve series name parsing support for Japanese. Thanks. 

(the reason is parses like that is because it doesn't know what the series is as there is no support. So hopefully after this, you'll get more expected results)
